soc/arm/common/cortex_m/arm_mpu_regions.c

```diff
@@ -21,13 +24,20 @@ static const struct arm_mpu_region mpu_regions[] = {
 #endif
 	/* Region 1 */
 	MPU_REGION_ENTRY("SRAM_0",
+#ifdef USE_PARTITION_MANAGER
```

Comment:
A sort of generic solution would be to use linker symbols here, since we, effectively, want to configure the SRAM used by Zephyr image. This could work upstream as well. Please investigate.

Reply:
AFAICS we cannot use linker symbols as value is required by the preprocessor.

Reply:
Maybe you could change the usage of CONFIG_SRAM_* to use linker values instead, or do they need to be macros? That way you don't need the new ifdef.